accepted/2024/net-standard-recommendation.md

```diff
@@ -0,0 +1,128 @@
+implementations that are still supported, can reference .NET Standard 2.0
+libraries.
+
+However, we have found that many customers still target .NET Standard 1.x. We
+found out the reasons boil down to one of two things:
+
+1. Targeting the lowest possible versions is considered better
+2. It was already like that and there is no reason to change
+
+The first argument has merit in a world where .NET Standard continues to produce
+new versions. Since that's no longer the case you might as well update to a
+later version that allows you to use more functionality. We found the biggest
+jump in productivity is with .NET Standard 2.0 because it also supports
+[referencing .NET Framework libraries][netfx-compate-mode], which is useful in
+cases where you modernize existing applications and have some mixture until you
+can upgrade all the projects.
+
+The second reason is always valid; however, in cases when you're actively
+maintaining a code base it makes sense to upgrade to avoid missing out on
+features that could have made your life easier and you didn't realize exist
+because code completion simply didn't advertise them.
+
+## Scenarios and User Experience
+
+### Building a project targeting .NET Standard
+
+Jackie is assigned a work item to add a feature that requires writing code in
+the shared business logic project. That project happens to target .NET Standard
+1.4. She doesn't know why because the project was built before her time on the
+team.  The first time she compiles in Visual Studio she sees a warning:
+
+> warning NETSDK1234: Contoso.BusinessLogic.csproj: Targeting .NET Standard
+> prior to 2.0 is no longer recommended. See \<URL\> for more details.
+
+Following the link, she finds a page that inform her that all supported .NET
+implementations allow using .NET Standard 2.0 or higher. She asks one of her
+colleagues and she remembers that they used .NET Standard 1.4 because at some
+point her team also maintained a Windows Phone app (which, to everybody's
+disappointment no longer exists).
+
+Jackie decides to upgrade the project to .NET Standard 2.0. She later realizes
+that this was a good idea because it enables her to use many of the
+`Microsoft.Extensions` libraries that make her life a lot easier, specifically
+configuration and DI.
+
+## Requirements
+
+### Goals
+
+* Issue a warning when *building* a project targeting `netstandard1.x` with the
+  .NET 9 SDK
+
+### Non-Goals
+
+* Issue a warning when building a project targeting `netstandard2.x`
+* Issue a warning when consuming a library that was built for .NET Standard 1.x.
+* Issue a warning building with a .NET SDK prior to 9.0
+
+## Stakeholders and Reviewers
+
+* Libraries team
+* SDK team
+* Build team
+* NuGet team
+
+## Design
+
+The .NET SDK will check the `TargetFramework` property. If it uses
+`netstandardX.Y` with `X.Y < 2.0` it will produce the following warning:
+
+> Targeting .NET Standard prior to 2.0 is no longer recommended. See \<URL\> for
```

Comment:
Does this need a URL? What about something like:

> .NET Standard 1.X is no longer recommended. Please upgrade to .NET Standard 2.0 or higher. For example: <TargetFramework>netstandard2.0</TargetFramework>

Reply:
I think we'll want a URL to explain what this warning means and when it's OK to suppress; we've done this or all other `NETSDKXXXX` warnings/errors.